### PR TITLE
Preview Spatial: Adds Feature and FeatureCollection

### PR DESCRIPTION
**Pull request by bchong95**
# Preview Spatial: Adds Feature and FeatureCollection

## Description

Adds `Feature` and `FeatureCollection` to the spatial object models. This should allow users to incorporate these types into their POCOs for insertion and reads. They can also be used for LINQ query translation.

### CODE REVIEW

**Review by sboshra — APPROVED** [review]
:shipit:

File: Microsoft.Azure.Cosmos/src/Spatial/FeatureCollection.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    internal
+#endif
+    sealed class FeatureCollection
+    {
+        public FeatureCollection(IReadOnlyList<Feature> features)
+        {
+            this.Features = features ?? throw new ArgumentNullException(nameof(features));
+        }
+
+        [DataMember(Name = "features")]
```
Comment (bchong95) [inline Microsoft.Azure.Cosmos/src/Spatial/FeatureCollection.cs:25]: missing the type field

File: Microsoft.Azure.Cosmos/src/Spatial/Feature.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.Cosmos.Spatial
+{
+    using System.Collections.Generic;
+    using System.Runtime.Serialization;
+    using Newtonsoft.Json;
+
+    [DataContract]
+#if PREVIEW
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Spatial/Feature.cs:12]: FYI all public preview feature are gated on Java availability. 
Spatial are not yet in Java so good to go.

**Review by kirankumarkolli — DISMISSED** [review]

File: Microsoft.Azure.Cosmos/src/Spatial/Feature.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+            this.Type = type == GeoJsonType.Feature ? type : throw new ArgumentException($"{type} must be {GeoJsonType.Feature}");
+            this.Geometry = geometry;
+            this.Properties = properties;
+            this.Id = id;
+        }
+
+        [DataMember(Name = "id")]
+        [JsonProperty("id", NullValueHandling = NullValueHandling.Ignore)]
+        public JToken Id { get; }
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/Spatial/Feature.cs:56]: >JToken [](start = 15, length = 6)

object

File: Microsoft.Azure.Cosmos/src/Spatial/Feature.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        [JsonConverter(typeof(StringEnumConverter))]
+        public GeoJsonType Type { get; }
+
+        [DataMember(Name = "geometry")]
+        [JsonProperty("geometry", Required = Required.AllowNull)]
+        public Geometry Geometry { get; }
+
+        [DataMember(Name = "properties")]
+        [JsonProperty("properties", Required = Required.AllowNull)]
+        public JObject Properties { get; }
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/Spatial/Feature.cs:69]: >JObject [](start = 15, length = 7)

Dictionary

**Review by sboshra — APPROVED** [review]
:shipit:

**Comment by j82w** [comment]
Closing since there has been no update for over 30 days. Please re-open if necessary.

**Pull request closed without merging**